### PR TITLE
scripts for some AI refactor

### PR DESCRIPTION
**Pull request by Reggie1998**
Some enemy controller things which make it better and a bit more manageable.

Through the root/animation movement combo would make it smooth, but it's still slippery albeit it actually plays and uses the animations decently now

### CODE REVIEW

File: Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs [hunk 1]
```diff
@@ -36,6 +39,7 @@ void Update()
 
     public void CallCombatAction(EnemyCombatAction actionData, GameObject player)
     {
+        Debug.Log(actionData.name);
```
Comment (ninjamelons) [inline Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs:42]: get that shit outta here

File: Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs [hunk 2]
```diff
@@ -72,40 +76,52 @@ public void CallCombatAction(EnemyCombatAction actionData, GameObject player)
 
     public GameObject GetIndicator(string abilityName)
     {
-        return attackIndicators?[abilityName];
+        if (attackIndicators.ContainsKey(abilityName)) {
+            return attackIndicators?[abilityName];
```
Comment (ninjamelons) [inline Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs:80]: "?" isnt really needed here anymore is it?

File: Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs [hunk 3]
```diff
@@ -125,33 +141,41 @@ public void ProjectileAttack(EnemyCombatAction actionData, GameObject player)
...
-                StartCoroutine(MoveTowardsTarget(5.0f, direction2, 4.9f, currentProjectile));
+                StartCoroutine(MoveTowardsTarget(12.0f, direction2, 4.9f, currentProjectile));
                 if (currentProjectile) {
                     StartCoroutine(GenericMonoHelper.Instance.DestroyAfterDelay(5.0f, currentProjectile));
                 }
 
             });
+
+            CallMethodWithDelay(actionData.castTime, () => {
+                //TODO: unsure when to destroy the projectile might need a function to check when it hits terrain/walls or reaches max range
```
Comment (ninjamelons) [inline Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs:168]: Make github issue or add to Miro board

File: Assets/AASourcCream/Scripts/EnemyNPC/EnemyAnimationCtrl.cs [hunk 4]
```diff
@@ -15,72 +15,58 @@ public class EnemyAnimationCtrl : SerializedMonoBehaviour
...
+        // if (!once) {
+        //     _animator.Play("Goblin_slash", -1);
+        //     //_animator.Play("death", -1);
+        //     Debug.Log(LayerMask.NameToLayer("Base Layer"));
+        //     once = true;
+        // }
     }
 
-    public void changeAnimationState(string newState)
+    //TODO use this https://docs.unity3d.com/ScriptReference/Animator.CrossFade.html crossFade seems to be smooth transition playing
```
Comment (ninjamelons) [inline Assets/AASourcCream/Scripts/EnemyNPC/EnemyAnimationCtrl.cs:56]: Miro or issue

**Review by ninjamelons — APPROVED** [review]

File: Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs (hunk 2 above)
Comment (Reggie1998) [reply Assets/AASourcCream/Scripts/EnemyNPC/AbilityController.cs]: yes, maybe would be reasonable to do attackIndicators?.ContainsKey instead now, but seems worthless

**Pull request merged**